### PR TITLE
Minor quirk system changes, adds on_addiction_removal proc

### PR DESCRIPTION
**Pull request by Ms-Mee**
## About The Pull Request
This PR adds quirk species-locks, fixes the fact that quirk conflicts were never properly acccounted for and adds on_addiction_remove, makes quirk traits a list instead of one singular, allowing for additional traits on a quirk, makes new players no longer be forced to have Navigationally Challenged(still available to take as a neutral quirk).
Still a bit messy so it's drafted, look at the files changed if you wish to see why.

## Why It's Good For The Game
Species-locked quirks are cool, conflicts not being accounted for is bad, navigationally challenged doesn't work enough to be forced onto new players I don't think.

## Changelog
:cl:
add: Added quirk species-locks, as well as a proc for effects that occur when an addiction is removed.
del: Navigationally challenged is no longer forcefully assigned to new players.
refactor: Refactored quirk conflicts, changed quirk_traits to a list, last step in quirk subsystem initialization is making all current clients have their quirklist checked for conflicts.
/:cl:

### CODE REVIEW

File: code/datums/traits/_quirk.dm [hunk 1]
```diff
@@ -9,7 +9,8 @@
 	var/lose_text
 	var/medical_record_text //This text will appear on medical records for the trait. Not yet implemented
 	var/mood_quirk = FALSE //if true, this quirk affects mood and is unavailable if moodlets are disabled
-	var/mob_trait //if applicable, apply and remove this mob trait
+	var/list/mob_traits //if applicable, apply and remove these mob traits
+	var/list/species_lock = list() //List of id-based locks for species, set "type" = "allowed" to lock the species not on the list, set "type" = "blocked" to lock the species on the list, then add the species ids to the list. Example: species_lock = list("type" = "allowed", "ipc", "plasmaman")
```
Comment (francinum) [inline code/datums/traits/_quirk.dm]: While this system is clever, you should probably define macros for this
```
list(
  TRAIT_SPECIES_WHITELIST
)
list(
  TRAIT_SPECIES_BLACKLIST
)
```

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -1284,47 +1284,121 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+	var/list/quirk_conflicts = list()
+	var/list/handled_conflicts = list()
+	for(var/quirk_index in SSquirks.quirks)
+		var/datum/quirk/quirk_datum = SSquirks.quirks[quirk_index]
+		var/datum/quirk/quirk_instance = new quirk_datum
+		if(quirk_instance.mood_quirk && CONFIG_GET(flag/disable_human_mood))
+			quirk_conflicts[quirk_instance.name] = "Mood is disabled."
+			if(!handled_conflicts["mood"])
+				handle_quirk_conflict("mood", null, user)
+				handled_conflicts(["mood"] = TRUE
```
Comment (Ms-Mee) [inline code/modules/client/preferences.dm]:
```suggestion
				handled_conflicts["mood"] = TRUE
```

**Comment by Ms-Mee** [comment]
Linters what fucking non-bitwise << use are you on about

**Comment by Ms-Mee** [comment]
> Linters what fucking non-bitwise << use are you on about

I guess maybe the to_chat world? I suppose it isn't *necessary*

**Pull request closed without merging**